packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -2133,17 +2139,22 @@ def process_event(
         # if that's the case, we proceed with the termination transition function,
         # regardless of what the current round is
         if self.is_termination_set and event == self.termination_event:
+            self.termination_transition_function = cast(
+                AbciAppTransitionFunction, self.termination_transition_function
+            )
+            self.background_round_cls = cast(AppState, self.background_round_cls)
             next_round_cls = self.termination_transition_function[
                 self.background_round_cls
             ].get(event, None)
```

Comment:
we only enter here is event == termination_event, so what is the reason to have an optional `next_round_cls` here?

Reply:
That was the definition before this change. I didn't see a reason to change it.